docs/README.md

```diff
@@ -0,0 +1,241 @@
+# Documentation
+
+----------------------------------------------
+
+## Building
+
+Building the documentation is pretty easy to do but it does have some requirements
+That have to be filled prior to building them.
+
+Here are the requirements:
+
+* Doxygen
+* Python >= 3.10
+* C compiler (gcc, msvc, clang, etc...)
+
+
+There are also some Python specific libraries that need to be installed. 
+You can either install these individually or you can use pip to read the requirements
+file to install everything that is needed for Python.
+
+* Sphinx
+* breathe
+* imagesize
+* importlib-metadata
+* sphinx-rtd-theme
+* sphinx-sitemap
+* sphinxcontrib-applehelp
+* sphinxcontrib-devhelp
+* sphinxcontrib-htmlhelp
+* sphinxcontrib-jsmath
+* sphinxcontrib-qthelp
+* sphinxcontrib-serializinghtml
+* sphinxcontrib-mermaid
+* sphinx-design
+* sphinx-rtd-dark-mode
+* typing-extensions
+
+To install using the `requirements.txt` file use the following command.
+
+    pip install -r requirements.txt
+
+
+Once you have all of the requirements installed you are ready to build them.
+To build the documentation use the following command.
+
+    python build.py skip_latex clean
+
+You may have to use the following command if you are un a Unix like OS
+
+    python3 build.py skip_latex clean
+
+The documentation will be output into the folder `out_html` in the root directory 
+for LVGL.
+
+## For Developers
```

Comment:
```suggestion
## .rst cheat sheet
```

Reply:
It is set as for developers because doxygen syntax going to be added to that section as well.